File: src/main/kotlin-graphql/io/github/graphql/model/RateLimitTO.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-06T19:48:04+0800"
)
open class RateLimitTO(
    val cost: Int,
    val limit: Int,
    val nodeCount: Int,
    val remaining: Int,
    val resetAt: String
) {

    companion object {
        fun builder(): Builder = Builder()
    }

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        joiner.add("cost: " + GraphQLRequestSerializer.getEntry(cost))
        joiner.add("limit: " + GraphQLRequestSerializer.getEntry(limit))
        joiner.add("nodeCount: " + GraphQLRequestSerializer.getEntry(nodeCount))
        joiner.add("remaining: " + GraphQLRequestSerializer.getEntry(remaining))
        joiner.add("resetAt: " + GraphQLRequestSerializer.getEntry(resetAt))
        return joiner.toString()
    }

    class Builder {

        private var cost: Int = 0
        private var limit: Int = 0
        private var nodeCount: Int = 0
        private var remaining: Int = 0
        private lateinit var resetAt: String

        fun setCost(cost: Int): Builder {
            this.cost = cost
            return this
        }

        fun setLimit(limit: Int): Builder {
            this.limit = limit
            return this
        }

        fun setNodeCount(nodeCount: Int): Builder {
            this.nodeCount = nodeCount
            return this
        }

        fun setRemaining(remaining: Int): Builder {
            this.remaining = remaining
            return this
        }

        fun setResetAt(resetAt: String): Builder {
            this.resetAt = resetAt
            return this
        }

        fun build(): RateLimitTO {
            return RateLimitTO(cost, limit, nodeCount, remaining, resetAt)
        }
    }
}

File: src/main/scala-graphql/io/github/graphql/model/RateLimitTO.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:01+0800"
)
class RateLimitTO(
    cost: Int,
    limit: Int,
    nodeCount: Int,
    remaining: Int,
    @javax.validation.constraints.NotNull
    resetAt: String
) {

    override def toString(): String = {
        Seq(
            "cost: " + GraphQLRequestSerializer.getEntry(cost),
            "limit: " + GraphQLRequestSerializer.getEntry(limit),
            "nodeCount: " + GraphQLRequestSerializer.getEntry(nodeCount),
            "remaining: " + GraphQLRequestSerializer.getEntry(remaining),
            if (resetAt != null) "resetAt: " + GraphQLRequestSerializer.getEntry(resetAt) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }
}

object RateLimitTO {

    def builder(): RateLimitTO.Builder = new Builder()

    class Builder {

        private var cost: Int = _
        private var limit: Int = _
        private var nodeCount: Int = _
        private var remaining: Int = _
        private var resetAt: String = _

        def setCost(cost: Int): Builder = {
            this.cost = cost
            this
        }

        def setLimit(limit: Int): Builder = {
            this.limit = limit
            this
        }

        def setNodeCount(nodeCount: Int): Builder = {
            this.nodeCount = nodeCount
            this
        }

        def setRemaining(remaining: Int): Builder = {
            this.remaining = remaining
            this
        }

        def setResetAt(resetAt: String): Builder = {
            this.resetAt = resetAt
            this
        }

        def build(): RateLimitTO = new RateLimitTO(cost, limit, nodeCount, remaining, resetAt)

    }
}
